astropy/modeling/utils.py

```diff
@@ -484,3 +484,64 @@ def combine_labels(left, right):
 
     return left + right
 
+
+def render_models2sky(model_list, image, x_key='x_mean', y_key='y_mean'):
+    """
+    Evaluates a list of 2D models on an 2D image array. Limits the evaluation to the
+    bounding box limits if present. 
+    """
+    #from .. nddata.utils import add_array
+
+    # Get position param names if not input.
+    # Try known variants, throw an error if not reconized.
+    # Better way to do this?
+    # Deal with this later, considering only Gaussian2D for now.
+    # for pos_key in ('mean', 'pos', '0'):
+    #     try:
+    #         x_pos = model.parameters[model.param_names.index('x_' + pos_key)]
+    #         y_pos = model.parameters[model.param_names.index('y_' + pos_key)]
+    #         break
+    #     except:
+    #         pass
+
+    # try:
+    #     x_pos, y_pos
+    # except:
+    #     raise InputParameterError, 'position keywords not found.'
+    
+    image = image.copy()
+    im_shape = image.shape
+
+
+    for model in model_list:
+
+        assert (model.n_inputs == 2) & (image.ndim == 2)
+
+        params = dict([(n, p) for n, p in zip(model.param_names, model.parameters)])
+
+        x_pos = int(params[x_key])
+        y_pos = int(params[y_key])
+
+        try:
+
+            bbox = model.bbox
+            assert bbox is not None
+            bbox = (max([0,int(bbox[0])]), min([im_shape[1]-1,int(bbox[1])]),
+                               max([0,int(bbox[2])]), min([im_shape[0]-1,int(bbox[3])]))
+            x, y = np.meshgrid(range(bbox[0],bbox[1]),
```

Comment:
After thinking about this a bit more, I guess I prefer the solution, where the coordinate arrays, that are used to evaluate the model, are created right at the beginning. You could just use `extract_array` to cut out the corresponding part that corresponds to your bounding box and evaluate the model with that. Again the handling of boundaries, sub-pixel positions and n-dimensional data comes for free...